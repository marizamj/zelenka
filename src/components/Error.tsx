/** @jsx jsx */
import { jsx, css } from "@emotion/core";
import { breakpoints } from "../lib/misc";

const Error = () => (
  <div css={errorStyle} data-testid="error">
    Что-то пошло не так. Пожалуйста, напишите нам на почту{" "}
    <a href="mailto:zelenkanasha@gmail.com">zelenkanasha@gmail.com</a> или в{" "}
    <a
      href="https://www.facebook.com/3elenka.online/"
      target="_blank"
      rel="noopener noreferrer"
    >
      фейсбук
    </a>
    .
  </div>
);

const errorStyle = css`
  color: #c92121;
  text-align: center;
  margin: auto;
  margin-top: 1em;
  font-size: 1.5em;
  max-width: 800px;
  padding: 1em;

  a {
    color: #c92121;
  }

  ${breakpoints.maxWidth1023} {
    font-size: 1.3em;
  }
`;

export default Error;
