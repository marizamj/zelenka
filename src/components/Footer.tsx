/** @jsx jsx */
import { jsx, css } from "@emotion/core";

const Footer = ({ currentYear }: { currentYear?: number }) => (
  <footer css={footerStyle} data-testid="footer">
    &#169; 2018-{currentYear || new Date().getFullYear()} zelenka.online{" "}
    <a href="mailto:zelenkanasha@gmail.com">zelenkanasha@gmail.com</a>
  </footer>
);

const footerStyle = css`
  border-top: 1px solid #398a85;
  padding: 1em;
  margin-top: 2em;
`;

export default Footer;
