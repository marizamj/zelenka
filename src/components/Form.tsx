/** @jsx jsx */
import { jsx, css } from "@emotion/core";
import { useState } from "react";
import formFields from "../lib/formFields";
import { breakpoints } from "../lib/misc";
import { FormType, FormData, FormFull } from "../types";

interface Props {
  type: FormType;
  handleSubmit: (formState: FormData) => void;
}

const Form = ({ type, handleSubmit }: Props) => {
  const [formState, setFormState] = useState<FormFull>({} as FormFull);

  return (
    <form
      css={formStyle}
      onSubmit={e => {
        e.preventDefault();
        handleSubmit(formState);
      }}
      data-testid={`form-${type}`}
    >
      <div css={formDescriptionStyle}>{formFields[type].description}</div>
      <p css={requiredStyle}>* - обязательные поля</p>
      {formFields[type].fields.map(({ label, required, name, type }) => (
        <label key={label} htmlFor={`field-${name}`}>
          <p css={formLabelStyle}>
            {label}
            {required && <span css={requiredStyle}>*</span>}
          </p>
          {type === "textarea" ? (
            <textarea
              onChange={({ target: { value, name } }) => {
                setFormState({ ...formState, [name]: value });
              }}
              required={required}
              name={name}
              id={`field-${name}`}
              data-testid={`field-${name}`}
              value={formState[name] || ""}
            />
          ) : (
            <input
              onChange={({ target: { value, name } }) => {
                setFormState({ ...formState, [name]: value });
              }}
              required={required}
              type={type || "text"}
              name={name}
              id={`field-${name}`}
              data-testid={`field-${name}`}
              value={formState[name] || ""}
            />
          )}
        </label>
      ))}
      <input type="submit" value="Отправить" data-testid="submit" />
    </form>
  );
};

const formStyle = css`
  font-size: 1.5em;
  width: 50%;
  margin-left: 25%;
  text-align: center;
  input,
  textarea {
    width: calc(100% - 20px);
    padding: 10px;
    outline: none;
    border: 1px solid lightgrey;
    border-radius: 3px;
    font-size: 0.8em;
  }
  textarea {
    height: 4em;
    resize: none;
  }
  input:focus,
  textarea:focus {
    border: 1px solid grey;
  }
  input[type="submit"] {
    cursor: pointer;
    color: dimgrey;
    font-size: 1em;
    width: auto;
    border: 1px solid lightgrey;
    border-radius: 3px;
    text-align: center;
    margin-top: 1em;
  }
  input[type="submit"]:hover {
    background: #f5f5f5;
  }
  input:disabled {
    background: lightgrey;
    cursor: default;
  }
  ${breakpoints.maxWidth1023} {
    width: 80%;
    margin: 0;
    margin-left: 10%;
    font-size: 1.3em;
  }
`;

const requiredStyle = css`
  font-size: 0.8em;
  color: #c92121;
  text-align: end;
  margin-left: 0.1em;
`;

const formDescriptionStyle = css`
  text-align: center;
`;

const formLabelStyle = css`
  color: dimgrey;
  margin-top: 1.5em;
  margin-bottom: 0.5em;
  font-size: 0.8em;
  text-align: start;
`;

export default Form;
