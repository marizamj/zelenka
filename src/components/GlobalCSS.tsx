/** @jsx jsx */
import { jsx, css, Global } from "@emotion/core";

const GlobalCSS = () => <Global styles={globalStyle} />;

const globalStyle = css`
  body {
    margin: 0;
    padding: 0;
    font-family: "PT Sans", Arial, sans-serif;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
    min-width: 320px;
  }

  code {
    font-family: source-code-pro, Menlo, Monaco, Consolas, "Courier New",
      monospace;
  }

  h1,
  h2 {
    color: #c92121;
    margin: 100px 0px 50px 0px;
  }
`;

export default GlobalCSS;
