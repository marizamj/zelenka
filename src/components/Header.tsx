/** @jsx jsx */
import { jsx, css } from "@emotion/core";
import { breakpoints } from "../lib/misc";

import logo from "../images/logo.png";
import topBottle from "../images/top-bottle.png";

const Header = () => (
  <header css={headerStyle}>
    <div css={titleImgStyle} />
    <div css={logoContainerStyle}>
      <img src={logo} css={logoStyle} alt="ЗЕЛЕНКА" />
      <p css={logoTextStyle}>
        Показываем хирургию и медицину так, чтобы не тошнило
      </p>
    </div>
  </header>
);

const headerStyle = css`
  position: relative;
  height: 150px;
  ${breakpoints.maxWidth1023} {
    z-index: 10;
  }
`;

const titleImgStyle = css`
  position: absolute;
  left: -50px;
  width: 400px;
  height: 400px;
  background: url(${topBottle});
  background-repeat: no-repeat;
  background-size: contain;
  ${breakpoints.maxWidth1023} {
    width: 350px;
    height: 350px;
    top: -125px;
  }
`;

const logoContainerStyle = css`
  text-align: center;
  position: absolute;
  width: 100%;
  top: 3em;
`;

const logoTextStyle = css`
  text-shadow: 0px 0px 1px #ffffff;
  font-size: 1.5em;
  font-weight: bold;
  margin: 10px;
  text-align: center;
  color: #c92121;
  ${breakpoints.maxWidth1023} {
    font-size: 1.3em;
  }
`;

const logoStyle = css`
  width: 300px;
  ${breakpoints.maxWidth1023} {
    width: 250px;
  }
`;

export default Header;
