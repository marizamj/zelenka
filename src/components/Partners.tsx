/** @jsx jsx */
import { jsx, css } from "@emotion/core";
import { useState } from "react";
import partners from "../lib/partners";
import { reactKey, breakpoints } from "../lib/misc";

const Partners = () => {
  const [visibleName, setVisibleName] = useState("");

  return (
    <div css={partnersStyle}>
      <h2>Уже работают с нами</h2>

      {partners.map(({ id, title, url, img }) => (
        <div
          css={partnerContainerStyle}
          key={reactKey({ id, title, url, img })}
        >
          <span css={partnerNameStyle(visibleName, id)}>{title}</span>
          <a href={url} target="_blank" rel="noopener noreferrer">
            <img
              src={img}
              alt={title}
              css={partnerLogoStyle}
              onMouseEnter={() => setVisibleName(id)}
              onMouseLeave={() => setVisibleName("")}
            />
          </a>
        </div>
      ))}
    </div>
  );
};

const partnersStyle = css`
  text-align: center;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-evenly;
  h2 {
    flex: 1 0 100%;
    margin-bottom: 1em;
  }
  ${breakpoints.maxWidth1023} {
    h2 {
      margin-bottom: 2em;
    }
  }
  ${breakpoints.maxWidth700} {
    display: initial;
  }
`;

const partnerContainerStyle = css`
  position: relative;
  padding: 4em 2em;
`;

const partnerNameStyle = (visibleName: string, id: string) => css`
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  font-size: 1em;
  opacity: ${visibleName === id ? 1 : 0};
  transition: opacity 0.2s ease;
  ${breakpoints.maxWidth1023} {
    opacity: 1;
  }
`;

const partnerLogoStyle = css`
  cursor: pointer;
  height: 130px;
`;

export default Partners;
