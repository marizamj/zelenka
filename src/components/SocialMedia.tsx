/** @jsx jsx */
import { jsx, css } from "@emotion/core";
import smBackImg from "../images/sm-back.png";
import facebook from "../images/facebook.png";
// import instagram from "../images/instagram.png";
import youtube from "../images/youtube.png";
import { breakpoints } from "../lib/misc";

const SocialMedia = () => (
  <div css={smStyle}>
    <a
      href="https://www.youtube.com/channel/UCIxnXifkJRTPOIe4AQM_fBQ"
      target="_blank"
      rel="noopener noreferrer"
    >
      <img
        src={youtube}
        alt="YouTube"
        css={smBtnStyle}
        style={{ marginRight: "5.5em" }}
      />
    </a>
    {/* <a
          href="https://www.youtube.com/channel/UCIxnXifkJRTPOIe4AQM_fBQ"
          target="_blank"
          rel="noopener noreferrer"
        >
          <img
            src={instagram}
            alt="Instagram"
            css={smBtnStyle}
            style={{ marginRight: "12vw" }}
          />
        </a> */}
    <a
      href="https://www.facebook.com/3elenka.online/"
      target="_blank"
      rel="noopener noreferrer"
    >
      <img src={facebook} alt="Facebook" css={smBtnStyle} />
    </a>
  </div>
);

const smStyle = css`
  font-size: 2em;
  position: relative;
  background: url(${smBackImg});
  background-repeat: no-repeat;
  background-size: contain;
  background-position: right;
  height: 250px;
  ${breakpoints.maxWidth1023} {
    font-size: 1.5em;
    height: 190px;
    margin-top: -20px;
    margin-left: 0;
  }
`;

const smBtnStyle = css`
  height: 2em;
  position: absolute;
  top: 130px;
  right: 0;
  margin-right: 3em;
  cursor: pointer;
  &:hover {
    opacity: 0.8;
  }
  ${breakpoints.maxWidth1023} {
    top: 100px;
  }
`;

export default SocialMedia;
