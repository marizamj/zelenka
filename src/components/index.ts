export { default as Error } from "./Error";
export { default as Form } from "./Form";
export { default as Header } from "./Header";
export { default as Partners } from "./Partners";
export { default as SocialMedia } from "./SocialMedia";
export { default as ThankYou } from "./ThankYou";
export { default as Video } from "./Video";
