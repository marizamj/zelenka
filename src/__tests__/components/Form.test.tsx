import React from "react";
import { render } from "@testing-library/react";

import { Form } from "../../components";

const mapToAttribute = (attribute: string) => (element: HTMLElement) =>
  element.getAttribute(attribute);

describe("Form", () => {
  test("Matches snapshot for partners", () => {
    const { container } = render(
      <Form type="partners" handleSubmit={jest.fn()} />
    );

    expect(container).toMatchSnapshot();
  });

  test("Matches snapshot for all", () => {
    const { container } = render(<Form type="all" handleSubmit={jest.fn()} />);

    expect(container).toMatchSnapshot();
  });

  test("Renders correct fields for partners", () => {
    const { getAllByTestId } = render(
      <Form type="partners" handleSubmit={jest.fn()} />
    );

    const fields = getAllByTestId(/^field-/);

    expect(fields).toHaveLength(7);
    expect(fields.map(mapToAttribute("name"))).toEqual([
      "name",
      "email",
      "phone",
      "workplace",
      "message",
      "money",
      "startDate"
    ]);
  });

  test("Renders correct fields for all", () => {
    const { getAllByTestId } = render(
      <Form type="all" handleSubmit={jest.fn()} />
    );

    const fields = getAllByTestId(/^field-/);

    expect(fields).toHaveLength(6);
    expect(fields.map(mapToAttribute("name"))).toEqual([
      "name",
      "email",
      "phone",
      "messageall",
      "compliments",
      "recommendations"
    ]);
  });
});
