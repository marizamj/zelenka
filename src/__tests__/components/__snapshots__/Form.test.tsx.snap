// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Matches snapshot for all 1`] = `
<div>
  <form
    class="css-13tf4h1-Form"
    data-testid="form-all"
  >
    <div
      class="css-19lds2-Form"
    >
      Хотите что-то сказать? Предложить идею или помочь с экспертизой? Стать героем выпуска, поделившись своей историей столкновения с хирургами? Мы внимательно прочтем ваше письмо и ответим, если заинтересуемся.
    </div>
    <p
      class="css-zjqj7m-Form"
    >
      * - обязательные поля
    </p>
    <label
      for="field-name"
    >
      <p
        class="css-1xzd690"
      >
        Как вас зовут?
        <span
          class="css-430mmm"
        >
          *
        </span>
      </p>
      <input
        data-testid="field-name"
        id="field-name"
        name="name"
        required=""
        type="text"
        value=""
      />
    </label>
    <label
      for="field-email"
    >
      <p
        class="css-1xzd690"
      >
        Контактный e-mail
        <span
          class="css-430mmm"
        >
          *
        </span>
      </p>
      <input
        data-testid="field-email"
        id="field-email"
        name="email"
        required=""
        type="email"
        value=""
      />
    </label>
    <label
      for="field-phone"
    >
      <p
        class="css-1xzd690"
      >
        Телефон
        <span
          class="css-430mmm"
        >
          *
        </span>
      </p>
      <input
        data-testid="field-phone"
        id="field-phone"
        name="phone"
        required=""
        type="tel"
        value=""
      />
    </label>
    <label
      for="field-messageall"
    >
      <p
        class="css-1xzd690"
      >
        Ваша идея/предложение
        <span
          class="css-430mmm"
        >
          *
        </span>
      </p>
      <textarea
        data-testid="field-messageall"
        id="field-messageall"
        name="messageall"
        required=""
      />
    </label>
    <label
      for="field-compliments"
    >
      <p
        class="css-1xzd690"
      >
        Чем вам нравится Зелёнка?
      </p>
      <textarea
        data-testid="field-compliments"
        id="field-compliments"
        name="compliments"
      />
    </label>
    <label
      for="field-recommendations"
    >
      <p
        class="css-1xzd690"
      >
        Что нам необходимо улучшить в своей работе?
      </p>
      <textarea
        data-testid="field-recommendations"
        id="field-recommendations"
        name="recommendations"
      />
    </label>
    <input
      data-testid="submit"
      type="submit"
      value="Отправить"
    />
  </form>
</div>
`;

exports[`Form Matches snapshot for partners 1`] = `
<div>
  <form
    class="css-13tf4h1-Form"
    data-testid="form-partners"
  >
    <div
      class="css-19lds2-Form"
    >
      Если вы врач, представитель клиники или любой другой компании, связанной с медициной, мы можем сделать очередной выпуск вместе. Заполните анкету, и мы очень скоро свяжемся с вами.
    </div>
    <p
      class="css-zjqj7m-Form"
    >
      * - обязательные поля
    </p>
    <label
      for="field-name"
    >
      <p
        class="css-1xzd690"
      >
        Как вас зовут?
        <span
          class="css-430mmm"
        >
          *
        </span>
      </p>
      <input
        data-testid="field-name"
        id="field-name"
        name="name"
        required=""
        type="text"
        value=""
      />
    </label>
    <label
      for="field-email"
    >
      <p
        class="css-1xzd690"
      >
        Контактный e-mail
        <span
          class="css-430mmm"
        >
          *
        </span>
      </p>
      <input
        data-testid="field-email"
        id="field-email"
        name="email"
        required=""
        type="email"
        value=""
      />
    </label>
    <label
      for="field-phone"
    >
      <p
        class="css-1xzd690"
      >
        Телефон
        <span
          class="css-430mmm"
        >
          *
        </span>
      </p>
      <input
        data-testid="field-phone"
        id="field-phone"
        name="phone"
        required=""
        type="tel"
        value=""
      />
    </label>
    <label
      for="field-workplace"
    >
      <p
        class="css-1xzd690"
      >
        Где вы работаете? Не забудьте адрес сайта.
        <span
          class="css-430mmm"
        >
          *
        </span>
      </p>
      <input
        data-testid="field-workplace"
        id="field-workplace"
        name="workplace"
        required=""
        type="text"
        value=""
      />
    </label>
    <label
      for="field-message"
    >
      <p
        class="css-1xzd690"
      >
        О чем вы хотели бы рассказать в Зелёнке?
        <span
          class="css-430mmm"
        >
          *
        </span>
      </p>
      <textarea
        data-testid="field-message"
        id="field-message"
        name="message"
        required=""
      />
    </label>
    <label
      for="field-money"
    >
      <p
        class="css-1xzd690"
      >
        Сколько денег вы готовы потратить на спонсорский выпуск?
      </p>
      <input
        data-testid="field-money"
        id="field-money"
        name="money"
        type="text"
        value=""
      />
    </label>
    <label
      for="field-startDate"
    >
      <p
        class="css-1xzd690"
      >
        Когда хотите начать?
      </p>
      <input
        data-testid="field-startDate"
        id="field-startDate"
        name="startDate"
        type="text"
        value=""
      />
    </label>
    <input
      data-testid="submit"
      type="submit"
      value="Отправить"
    />
  </form>
</div>
`;
